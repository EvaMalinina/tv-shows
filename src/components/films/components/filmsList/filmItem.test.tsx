import React from 'react';
import { render } from '@testing-library/react';
import FilmsItemC from './FilmItem';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('It should render list item - single movie view', () => {
  const mockStore = configureStore();
  const filmMockProps = {
    movieId: '12345',
    name: 'Hello',
    desc: 'I am hello',
    category: 'comedy',
    year: 2021,
    img: 'link',
    runtime: 100
  }
  const initialState = [];
  const store = mockStore(initialState.push(filmMockProps));

  const name = filmMockProps.name;

  const FilmsItem = (name: string) => {
    return <>{
      <Provider store={store}>
        name && <FilmsItemC
          movieId={filmMockProps.movieId}
          name={filmMockProps.name}
          desc={filmMockProps.desc}
          category={filmMockProps.category}
          year={filmMockProps.year}
          img={filmMockProps.img}
          runtime={filmMockProps.runtime}
        />
      </Provider>
    }</>
  }

  test('Render single movie view', () => {
    const { asFragment } = render(FilmsItem(name));

    expect(asFragment()).toMatchSnapshot();
  });

});