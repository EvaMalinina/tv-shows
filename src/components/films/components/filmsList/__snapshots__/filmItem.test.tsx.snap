// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render list item - single movie view Render single movie view 1`] = `
<DocumentFragment>
  name && 
  <div
    class="sc-pFZIQ kievCr"
  >
    <div
      class="sc-gsTCUz gATSVo"
    >
      <img
        src="link"
        style="width: 100%; height: 300px;"
      />
      <button
        class="sc-jrAGrp inVoEh"
      >
        :
      </button>
      <h4
        style="text-transform: capitalize;"
      >
        Hello
      </h4>
      <p>
        I am hello
      </p>
      <span>
        2021
      </span>
    </div>
  </div>
</DocumentFragment>
`;
